universal-application-tool-0.0.1/test/services/applicant/ApplicantServiceImplTest.java

```diff
@@ -160,6 +189,23 @@ public void stageAndUpdateIfValid_hasPathNotInBlockException() {
     assertThat(errorAnd.getErrors().asList().get(0)).isInstanceOf(PathNotInBlockException.class);
   }
 
+  @Test
+  public void stageAndUpdateIfValid_hasIllegalArgumentExceptionForReservedScalarKeys() {
+    Applicant applicant = subject.createApplicant(1L).toCompletableFuture().join();
+    String reservedScalar = "applicant.name." + QuestionDefinition.METADATA_UPDATE_TIME_KEY;
+    ImmutableMap<String, String> updates = ImmutableMap.of(reservedScalar, "12345");
+
+    ErrorAnd<ReadOnlyApplicantProgramService, Exception> errorAnd =
+        subject
+            .stageAndUpdateIfValid(applicant.id, programDefinition.id(), 1L, updates)
+            .toCompletableFuture()
+            .join();
```

Comment:
Suggestion (blocking): Please throw an exception in the method and catch it here. `join()` will throw an unchecked `CompletionException`.

Reply:
Thanks! I believe this is fixed